File: performance/string_operations.lua
math.randomseed(42)
local res = ""
local count = 0

for i = 0, 4999 do
  local num = math.random(0, 9)
  local str_num = tostring(num)

  res = res .. str_num .. ","
  count = count + #res
end

print(count)
print(res)
print(#res)

File: performance/string_operations.py
import random

def main():
    res = ""
    count = 0

    # String operations benchmark
    for i in range(5000):
        num = random.randint(0, 9)
        str_num = str(num)

        # String concatenation
        res = res + str_num + ","

        # String length operations
        count = count + len(res)

    print(count)
    print(res)
    print(len(res))

main()